elastic/eui
# [EUI+] Move all Elastic Charts docs from the `Components` tab to the `Data visualization`

## Summary

This PR moves our charts documentation that was previously in `Components` to the top-level `Data visualization` section, and reorganizes several sections to make more sense with the new page guideline intros.

It also opinionatedly renames the URL slug of `/data-viz/` to `/charts/` (to match the team/repo name?), although I'm very open to feedback on that.

<img width="1350" alt="" src="https://github.com/user-attachments/assets/1bd40acd-91bb-49d1-9263-978e554f0a7c">

## QA

- https://eui.elastic.co/pr_8182/new-docs/docs/charts/design-system/ - sanity check whether sidebar IA makes sense as well as general docs copy

### General checklist

N/A, docs only
